alternatives: update doc to use the usual `alternatives` commands (#673)
coreos/fedora-coreos-docs: modules/ROOT/pages/alternatives.adoc

= Setting alternatives

> Now Fedora CoreOS systems can use the usual `alternatives` commands to configure the default command, as https://github.com/fedora-sysv/chkconfig/issues/9[issue] is fixed.

Reviewer: Let's keep the previous content at the bottom as this still applies until everyone as moved to F41 based FCOS. Something like: "Starting with Fedora CoreOS based on Fedora 41, ...". The bottom section can be named "manual setup" or something like that.

> Now Fedora CoreOS systems can use the usual `alternatives` commands to configure the default command, as https://github.com/fedora-sysv/chkconfig/issues/9[issue] is fixed.

Reviewer: Let's move the fixed issue reference to the bottom or completely remove it as we don't reference all fixed issues in the doc.

For example, to use the legacy-based variants of the `iptables` commands:

> [source,bash,subs="attributes"]

Reviewer: Let's have a Butane example first and then a manual example using shell commands.

----
# The legacy priority is 10, update nft to lower priority, so legacy wins
pfx=/usr/sbin/iptables
pfx6=/usr/sbin/ip6tables

> sudo alternatives --install ${pfx} iptables ${pfx}-nft 5 \

Reviewer: Hum, that looks weird. I'll give it a try.

Author: Thanks for the kind review!

      overwrite: true
      hard: false
----

> == Manual setup

Reviewer: ```suggestion
== Manual configuration
```

Starting with Fedora CoreOS based on Fedora 41, you can use `alternatives` commands to configure the default command manually.

For example, to use the legacy-based variants of the `iptables` commands:

> .Example using alternatives to configure default command manually

Reviewer: ```suggestion
.Example using alternatives to configure default the iptables backend
```

      overwrite: true
      hard: false
----

== Using alternatives commands

Starting with Fedora CoreOS based on Fedora 41, you can use `alternatives` commands to configure the default command.

> . Example systemd unit to configure default the iptables backend

Reviewer: ```suggestion
. Example Butane config using a systemd unit to configure the default iptables backend
```

[source,yaml,subs="attributes"]
----
variant: fcos
version: {butane-latest-stable-spec}
systemd:
  units:
    - name: custom-iptables-nft.service
      enabled: true
      contents: |
        [Unit]
        Description=Set the nft backend for iptables 
        [Service]
        ExecStart=/usr/sbin/alternatives --set iptables /usr/sbin/iptables-nft
        RemainAfterExit=yes
        [Install]
        WantedBy=multi-user.target
----

You can also manually running `alternatives` commands to configure the default command runtime.

> .Example manually to configure default the iptables backend

Reviewer: ```suggestion
.Example to manually configure the default iptables backend
```

      overwrite: true
      hard: false
----

== Using alternatives commands

Starting with Fedora CoreOS based on Fedora 41, you can use `alternatives` commands to configure the default command.

. Example Butane config using a systemd unit to configure the default iptables backend
[source,yaml,subs="attributes"]
----
variant: fcos
version: {butane-latest-stable-spec}
systemd:
  units:
    - name: custom-iptables-default.service
      enabled: true
      contents: |
        [Unit]
        Description=Set the default backend for iptables 
        [Service]
        ExecStart=/usr/sbin/alternatives --set iptables /usr/sbin/iptables-legacy
        RemainAfterExit=yes
        [Install]
        WantedBy=multi-user.target
----

> NOTE: We don't recommend configuring the default iptables backend to `iptables-legacy`, it's just an example.

Reviewer 2: ```suggestion
NOTE: We don't recommend configuring the default iptables backend to `iptables-legacy`. This is just an example.
```

> You can also manually running `alternatives` commands to configure the default command runtime.

Reviewer 2: ```suggestion
You can also manually run the `alternatives` commands to configure the default command runtime.
```

. Example to manually configure the default iptables backend
[source,bash]
----
# Check the link info
alternatives --display iptables
iptables --version

> # Config iptables to point to iptables-nft

Reviewer 2: ```suggestion
# Configure iptables to point to iptables-nft
```
